src/game_battlealgorithm.cpp

```diff
@@ -627,22 +627,21 @@ void Game_BattleAlgorithm::AlgorithmBase::Apply() {
 
 	if (GetAffectedHp() != -1 && !was_dead) {
 		int hp = GetAffectedHp();
-		int target_hp = target->GetHp();
-		target->ChangeHp(IsPositive() ? hp : -hp, true);
+		hp = IsPositive() ? hp : -hp;
+		hp = target->ChangeHp(hp, true);
 		if (IsAbsorb() && !IsPositive()) {
 			// Only absorb the hp that were left
-			int src_hp = std::min(target_hp, hp);
-			source->ChangeHp(src_hp, true);
+			source->ChangeHp(-hp, false);
 		}
 	}
 
 	if (GetAffectedSp() != -1) {
 		int sp = GetAffectedSp();
-		int target_sp = target->GetSp();
-		target->SetSp(target->GetSp() + (IsPositive() ? sp : -sp));
+		sp = IsPositive() ? sp : -sp;
+		sp = target->SetSp(sp);
```

Comment:
Using `target->SetSp(sp)` unfortunately breaks SP changes. Changing this to `target->ChangeSp(sp)` fixes this. After this, everything should work then and I will approve this PR.

Reply:
That was a rebase error, fixed.